### PR TITLE
8572 Old Manage business dashboard updates

### PR DESCRIPTION
**Pull request by cameron-freshworks**
*Issue #:* /bcgov/entity#8572

*Description of changes:*
- Open button redirects to the Name Request app and queries Name Request for Display.

- Adds a button called "Use this name request" in drop down for Incorporations and this will launch draft IA. This button is not shown for unsupported NRs and BEN change of name/alteration (ie, only show for BEN IAs). Set Using a FF.

- "Remove" button relocated into Drop down menu.

 - We are using the combo button design. Open/drop down menu including remove and the 'Use this name Request' conditionally and "Remove"

 - Makes request to retrieve and populate Name Request data for any affiliation of NR Type.

 - Removes auto draft creation upon manually affiliating a name request and does so ONLY when selecting "Use this Name Request"

- Open action to open businesses regularly OR navigate to Name Request for NR affiliations.

- Set FF for controlling IA creations.

- When navigating to the NAME REQUEST - pass parameters required to pull up Existing NR Summary. ** stretch goal.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: auth-web/src/components/auth/manage-business/AffiliatedEntityList.vue
Comment (sumesh-aot) [inline auth-web/src/components/auth/manage-business/AffiliatedEntityList.vue]: shouldn't this be using the new attributes Thor added on NR GET response instead of checking against a configured value?

File: auth-web/src/components/auth/manage-business/AffiliatedEntityList.vue
Comment (sumesh-aot) [inline auth-web/src/components/auth/manage-business/AffiliatedEntityList.vue:32]: Does NR always sends the approved name as first one in array? If not, it cannot be assumed that the name is first one.

File: auth-web/src/components/auth/manage-business/AffiliatedEntityList.vue
Comment (severinbeauvais) [inline auth-web/src/components/auth/manage-business/AffiliatedEntityList.vue:175]: Is the default value falsey if LD can't get this flag (LD id not configured or network issue)?

File: auth-web/src/components/auth/manage-business/AffiliatedEntityList.vue
Comment (severinbeauvais) [inline auth-web/src/components/auth/manage-business/AffiliatedEntityList.vue:178]: Is Conditionally Approved a value state for using a NR?

File: auth-web/src/components/auth/manage-business/AffiliatedEntityList.vue
Comment (cameron-freshworks) [reply auth-web/src/components/auth/manage-business/AffiliatedEntityList.vue]: I thought i heard something about this but in a GET on `https://legal-api-dev.apps.silver.devops.gov.bc.ca/api/v1/nameRequests/<nrNumber>` we currently only do a proxy request through LEAR->NAMEX and back, i didn't see any additional properties. 
Where should i look for these?

File: auth-web/src/components/auth/manage-business/AffiliatedEntityList.vue
Comment (cameron-freshworks) [reply auth-web/src/components/auth/manage-business/AffiliatedEntityList.vue:32]: In this Dashboard (old), we will just be taking the first name, the approved name.
My understanding is that these Auto Affiliated NR's wont be published until APPROVED on their side, so this should do in the meantime.

File: auth-web/src/components/auth/manage-business/AffiliatedEntityList.vue
Comment (cameron-freshworks) [reply auth-web/src/components/auth/manage-business/AffiliatedEntityList.vue:175]: For sure, with optional chaining it would return `undefined` ie a falsey value in that event.

File: auth-web/src/components/auth/manage-business/AffiliatedEntityList.vue
Comment (cameron-freshworks) [reply auth-web/src/components/auth/manage-business/AffiliatedEntityList.vue:178]: I'm not sure, good question. Was not called out in the tickets🤷

File: auth-web/src/components/auth/manage-business/AffiliatedEntityList.vue
Comment (sumesh-aot) [reply auth-web/src/components/auth/manage-business/AffiliatedEntityList.vue]: Not sure if it is implemented yet. If not in place, I guess it's good to go with configured value.

File: auth-web/src/components/auth/manage-business/AffiliatedEntityList.vue
Comment (sumesh-aot) [reply auth-web/src/components/auth/manage-business/AffiliatedEntityList.vue:32]: Names publishes any state change to the NR and we add the NR to the account when the status is DRAFT, so it's not only APPROVED names but other states can also appear.
Also you may need to check the status as well, before you start a draft IA.

File: auth-web/src/components/auth/manage-business/AffiliatedEntityList.vue
Comment (cameron-freshworks) [reply auth-web/src/components/auth/manage-business/AffiliatedEntityList.vue]: Totally, i believe once it is implemented we would replace the first two properties but the third (feature flag) would most likely still be required for control on what we allow as per environment. 

Thanks for bringing this up, will look into these new properties and where to find them 👍

File: auth-web/src/components/auth/manage-business/AffiliatedEntityList.vue
Comment (cameron-freshworks) [reply auth-web/src/components/auth/manage-business/AffiliatedEntityList.vue:32]: Oh interesting, Andre said they were only publishing on APPROVED states. I'll look into it. 

But yes, we do look at State before enabling the ability to Create the Draft IA.

Comment (cameron-freshworks) [reply auth-web/src/components/auth/manage-business/AffiliatedEntityList.vue:32]: Will update this condition, Namex is passing the individual name state 👍

Comment (cameron-freshworks) [reply auth-web/src/components/auth/manage-business/AffiliatedEntityList.vue]: Updated to display the APPROVED name or the first name only in the event of a DRAFT.

Future state on the new dashboard WILL display all the draft/approved names, however that wasn't a requirement of the OLD dashboard.

File: auth-web/src/components/auth/manage-business/AffiliatedEntityList.vue
Comment (cameron-freshworks) [reply auth-web/src/components/auth/manage-business/AffiliatedEntityList.vue:178]: Will update this in the future if the requirements are updated 👍

**Review by sumesh-aot — APPROVED** [review]
LGTM

File: auth-web/src/components/auth/manage-business/AffiliatedEntityList.vue
Comment (severinbeauvais) [reply auth-web/src/components/auth/manage-business/AffiliatedEntityList.vue:178]: OK, I'll resolve this comment, but please ask BA if maybe this was a missed req.

File: auth-web/src/components/auth/manage-business/AffiliatedEntityList.vue
Comment (severinbeauvais) [reply auth-web/src/components/auth/manage-business/AffiliatedEntityList.vue:175]: OK, and you obviously handle that 👍

**Pull request merged**
